src/main/kotlin/app/revanced/patcher/util/Document.kt

```diff
@@ -34,15 +36,24 @@ class Document internal constructor(
                 readerCount.remove(it)
             }
 
-            it.outputStream().use { stream ->
-                TransformerFactory.newInstance()
-                    .newTransformer()
-                    .transform(DOMSource(this), StreamResult(stream))
+            val transformer = TransformerFactory.newInstance().newTransformer()
+            if (isAndroid) {
+                // Set to UTF-16 to prevent surrogate pairs from being escaped to broken numeric character references.
+                transformer.setOutputProperty(OutputKeys.ENCODING, "UTF-16")
+                // The XML declaration will have encoding="UTF-16", but we're going to write it back in UTF-8, so omit it.
+                transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes")
+                // Use FileWriter to output the XML file in UTF-8 encoding.
+                it.writer().use { writer ->
```

Comment:
```suggestion
                it.bufferedWriter().use { writer ->
```

I forgot about buffering
Just applying this suggestion is OK

Reply:
True, buffered writer is useful when doing many writes, and ToStream write char by char, thus it will improve perf.